File: 0849_maximize_distance_to_closest_person/max_dist_to_closest.java
class Solution {
    // Time: O(n)
    // Space: O(1)
    public int maxDistToClosest(int[] seats) {
        int res = 0, i = 0;
        int n = seats.length;
        while (i < n) {
            while (i < n && seats[i] == 1) {
                i++;
            }
            int j = i; // first pointer
            while (i < n && seats[i] == 0) {
                i++;
            }
            if (j == 0 || i == n) {
                res = Math.max(res, i - j);
            } else {
                res = Math.max(res, (i - j + 1) / 2);
            }
        }
        return res;
    }
}

File: 0849_maximize_distance_to_closest_person/max_dist_to_closest.py
from typing import List


def maxDistToClosest(seats: List[int]) -> int:
    """
    Time: O(n)
    Space: O(1)
    """
    res = i = 0
    n = len(seats)
    while i < n:
        while i < n and seats[i]:
            i += 1
        j = i
        while i < n and not seats[i]:
            i += 1
        if not j or i == n:
            res = max(res, i - j)
        else:
            res = max(res, (i - j + 1) // 2)
    return res
